### DISCUSSION TITLE
generate import map npx payload generate:importmap giving an error

### DISCUSSION THREAD

**Discussion by hassanbhatti1989** [payloadcms/payload#9861] [Q&A] [answered]
loco-payload@1.0.0 payload
> cross-env NODE_OPTIONS=--no-deprecation payload generate:importmap

node:internal/process/promises:394
    triggerUncaughtException(err, true /* fromPromise */);
    ^
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".css" for /Users/hassanbhatti/Desktop/loco-payload/node_modules/react-image-crop/dist/ReactCrop.css
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:218:9)
    at defaultGetFormat (node:internal/modules/esm/get_format:244:36)
    at defaultLoad (node:internal/modules/esm/load:122:22)
    at async nextLoad (node:internal/modules/esm/hooks:748:22)
    at async load (file:///Users/hassanbhatti/Desktop/loco-payload/node_modules/tsx/dist/esm/index.mjs?1733837167359:2:1762)
    at async nextLoad (node:internal/modules/esm/hooks:748:22)
    at async Hooks.load (node:internal/modules/esm/hooks:385:20)
    at async handleMessage (node:internal/modules/esm/worker:199:18) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}

Node.js v22.12.0

**Comment by aaronksaunders** [comment]
so this is my custom component, still a work in progress,
```typescript
'use client'

import { useEffect } from 'react'
import { useField } from '@payloadcms/ui'
import { useAllFormFields } from '@payloadcms/ui'
import { TextFieldClientComponent } from 'payload'
import { getSiblingData } from 'payload/shared'

export const CategoryDataField: TextFieldClientComponent = ({ path }) => {
  console.log('path ==>', path)

  const [fields, dispatchFields] = useAllFormFields()
  const categoryField = getSiblingData(fields, 'nominations.0.category')

  const { setValue, value } = useField({ path: 'nominations.0.categorySlug' })

  console.log('categoryField', value)

  useEffect(() => {
    if (categoryField?.category) {
      fetch(`/api/categories/${categoryField?.category}?depth=2`)
        .then((res) => res.json())
        .then((data) => {
          console.log('data', data)

          setValue(data?.categorySlug)
        })
    }
  }, [categoryField?.category, setValue])

  return null
}

```

and this is the Collection that uses it, just the relavant bits
```typescript
// imports
import { anyone } from '@/access/anyone'
import { authenticated } from '@/access/authenticated'
import { CollectionConfig } from 'payload'
import { CategoryDataField } from 'src/collections/Nominees/components/CategoryDataField'

// useage
        {
          name: 'categorySlug',
          type: 'text',
          virtual: true,
          admin: {
            readOnly: true,
            components: {
              Field: CategoryDataField as any,
            },
          },
        },
```

output from command
```
aaronksaunders@16-MacBook-Pro back-end-nia % npm run generate:importmap

> back-end-nia@1.0.0 generate:importmap
> cross-env NODE_OPTIONS=--no-deprecation payload generate:importmap


node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".css" for /Volumes/ExtremePro/work/NIA/back-end-nia/node_modules/react-image-crop/dist/ReactCrop.css
    at new NodeError (node:internal/errors:406:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:99:9)
    at defaultGetFormat (node:internal/modules/esm/get_format:142:36)
    at defaultLoad (node:internal/modules/esm/load:120:20)
    at nextLoad (node:internal/modules/esm/hooks:833:28)
    at load (file:///Volumes/ExtremePro/work/NIA/back-end-nia/node_modules/tsx/dist/esm/index.mjs?1733845947497:2:1768)
    at nextLoad (node:internal/modules/esm/hooks:833:28)
    at Hooks.load (node:internal/modules/esm/hooks:416:26)
    at handleMessage (node:internal/modules/esm/worker:168:24)
    at Immediate.checkForMessages (node:internal/modules/esm/worker:117:28) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}

Node.js v20.9.0
aaronksaunders@16-MacBook-Pro back-end-nia % 
```

**Reply by aaronksaunders** [reply]
Thanks, for help. I solved the issue with some help in discord, here is the correction for someone who might run into same issue
```typescript
  {
    name: 'categorySlug',
    type: 'text',
    virtual: true,
    admin: {
      readOnly: true,
      components: {
        Field: '/collections/Nominees/components/CategoryDataField#CategoryDataField',
      },
    },
  },
```
the name is added because it is a named export in my custom component and not a default export

https://payloadcms.com/docs/admin/components#component-paths

**Comment by hassanbhatti1989** [comment]
when i try to give url to custom component
payload version 3.5.0 and website template i am using

`{
      name:'customFaqs',
      type: 'array',
      label: 'Custom FAQs',
      defaultValue: [],
      fields: [
        {
          name: 'question',
          type: 'text',
          label: 'Question',
        },
        {
          name: 'answer',
          type: 'textarea',
          label: 'Answer',
        },
      ],
      admin: {
        // Specify your custom component here
        components: {
          Field: '@/components/FAQTabs',
        },
      },
    },`

[ Server ] getFromImportMap: PayloadComponent not found in importMap {key: "@/components/FAQTabs.tsx#default"PayloadComponent: "@/components/FAQTabs.tsx#default"schemaPath: "@/components/FAQTabs.tsx#default"} "You may need to run the `payload generate:importmap` command to generate the importMap ahead of runtime."

and when i try to run the command for generating npm install generate:importmap then i got error .css

**Comment by hassanbhatti1989** [comment]
`import` {CustomTimeField} from '@/components/TimeField'

export const ListingPosts: CollectionConfig<'posts'> = {
  slug: 'listings',
  admin: {
    group: 'Listings',
    hidden: false,
    useAsTitle: 'name',
    description: 'Contains all the listings for the site.',
    defaultColumns: ['name', 'price', 'category'],
    preview: (doc) => `/listings/${doc.slug}`,
    pagination: {
      defaultLimit: 10,
      limits: [10, 20, 50],
    },
    listSearchableFields: ['name', 'slug'],
  },
  fields: [
    {
      name:'customFaqs',
      type: 'array',
      label: 'Custom FAQs',
      defaultValue: [],
      fields: [
        {
          name: 'question',
          type: 'text',
          label: 'Question',
        },
        {
          name: 'answer',
          type: 'textarea',
          label: 'Answer',
        },
      ],
      admin: {
        // Specify your custom component here
        components: {
          Field: FAQTabs,
        },
      },
    },
  ]
}`


this is working but when i will try to give the direct path in strings
`
admin: {
        // Specify your custom component here
        components: {
          Field: '@/components/FAQTabs',
        },
      },
      `
then its not working and then its start to give an error

**Comment by akhrarovsaid** [comment, marked as answer]
For anyone else running into this, double check the `path` to your custom component. It should be a string path that points to where your custom component is. If it's a _named_ export, you should use a `#` followed by the variable name of the component. (IE: `'/components/MyComp#MyComp'`).

If you're getting a message in your logs that reads something along the lines of "importMap couldn't find the component," then it's most likely an issue with your path.

If you're getting a cryptic error message about file extensions, then I'd double check that you're not passing an actual React component to the path, and instead you should be passing a string path to it.

See [more on component paths.](https://payloadcms.com/docs/admin/components#component-paths)

**Comment by hassanbhatti1989** [comment]
solved it in my case its default so i need to add #default at the end of my path
Field: '/components/FAQTabs'#default,
